lib/tools/emacs/erlang.el

```diff
@@ -2068,33 +2161,54 @@ This function is aware of imported functions."
   (require 'man)
   (setq name (or name
                  (erlang-default-function-or-module)))
-  (let ((modname nil)
-        (funcname nil))
-    (cond ((string-match ":" name)
-           (setq modname (substring name 0 (match-beginning 0)))
-           (setq funcname (substring name (match-end 0) nil)))
-          ((stringp name)
-           (setq modname name)))
-    (when (or (null modname) (string= modname ""))
-      (error "No Erlang module name given"))
-    (cond ((fboundp 'Man-notify-when-ready)
-           ;; Emacs 19:  The man command could possibly start an
-           ;; asynchronous process, i.e. we must hook ourselves into
-           ;; the system to be activated when the man-process
-           ;; terminates.
-           (if (null funcname)
-               ()
-             (erlang-man-patch-notify)
-             (setq erlang-man-function-name funcname))
-           (condition-case err
-               (erlang-man-module modname)
-             (error (setq erlang-man-function-name nil)
-                    (signal (car err) (cdr err)))))
-          (t
-           (erlang-man-module modname)
-           (when funcname
-             (erlang-man-find-function (current-buffer) funcname))))))
-
+  (require 'cl)
+  (flet ((disp-man-page-erl-fun (name)
```

Comment:
erlang.el is currently declaring a dependency on Emacs 24.1 (near the top of the file). In Emacs 24.3 and above, this piece of code generates a compilation warning:

```
erlang.el:2164:13:Warning: ‘flet’ is an obsolete macro (as of 24.3); use
    either ‘cl-flet’ or ‘cl-letf’.
```

So if we were to increase the required version, we could change this to use `cl-flet` and drop the `(require 'cl)`. Perhaps it's time? Emacs 24.3 was released in 2013.

Reply:
I would prefer defun over any incarnation of flet (this works too, though).

Reply:
I have refactored the flet into a defun as @johanclaesson suggests (commit 
7864c56). This should also fix the warning issue in Emacs 24.3+ that @legoscia points out above. I have discussed upgrading the required Emacs version to 24.3 with @dgud. It is uncertain if such an upgrade will cause issues so we have to wait with upgrading until we have figured this out.

Reply:
We have done an investigation and concluded that we can upgrade the required Emacs version to 24.3. I have done so in commit 98cff9a (see below) and refactored away the flet expressions that existed in erlang-eunit.el and erlang-flymake.el. Please let me know if you can find any problems with the changes in erlang-eunit.el and erlang-flymake.el.